drivers/device_uid/Kconfig

```diff
@@ -0,0 +1,35 @@
+# Kconfig - UID driver configuration options
+
+#
+# Copyright (c) 2019 Alexander Wachter
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig DEVICE_UID
+	bool "Unique device id driver"
+	help
+	  Enable unique device id driver.
+
+if DEVICE_UID
+
+config UID_DEVICE_NAME
```

Comment:
Since this is not really a device, I'm not sure if we should use device tree here.
Regarding device info: I like this Idea! Maybe something like this:
device_info_get(struct device *dev, enum device_info_type, u8_t *buffer, size_t length);